doc/source/dev/contributor/continuous_integration.rst

```diff
@@ -0,0 +1,83 @@
+.. _continuous-integration:
+
+======================
+Continuous Integration
+======================
+
+Continuous integration is part of our development process and ensure that
+every piece of code or documentation which is contributed to SciPy is working
+and does not have unforeseen effects.
+
+.. note:: Before submitting or updating your PR, please ensure that you tested
+          locally your changes. See :ref:`pr-checklist`.
+
+Workflows
+=========
+
+We run more than 20 different workflows with different version of the
+dependencies, different architectures, etc. A PR must pass all these checks
+before it can be merged as to ensure a sustainable state of the project.
+
+Apart from the unit tests, the documentation and examples in the docstrings are
+also checked. These are common failing workflows as Sphinx and doctests have
+very strict rules. These aspects are very important as documentation and
+examples are user facing elements. Ensure that these elements are properly
+rendered.
+
+The logs can be long, but you will always find out why your did not pass a
+check. Simply click on ``Details`` to access the logs.
+
+Following is a list of all the different workflows in use. They are gouped
+by CI resources providers.
+
+GitHub Actions
+--------------
+* ``Linux Tests``: Python 3.7 and nightly Cpython 3.9
+* ``macOS Tests``: test matrix from Python 3.7 to 3.9
+
+Azure
+-----
+* ``Lint``: PEP8 and code style
+* ``Windows Python``: test matrix from Python 3.7 to 3.9
+* ``Linux_Python_37_32bit_full``
+* ``wheel_optimized_gcc48``
+* ``source_distribution``
+* ``refguide_asv_check``: doctests from examples
```

Comment:
... and benchmark suite

Reply:
or is the name wrong by now, and we only run ASV on CircleCI? if so, let's fix the name

Reply:
No we do something `python -masv check -E existing`. There is just no `displayName` (I will add one).